submissions/RomanSaiko/oop/index.js

```diff
@@ -0,0 +1,64 @@
+class Inhabitant {
+    constructor ({ species, name, gender, saying, friends = "No friends! One man army" }) {
+        this.species = species
+        this.name = name
+        this.gender = gender
+        this.saying = saying
+        this.friends = friends
+    }
+
+    toString() {
+        const props = Object.entries(this).map(([key, value]) => `${key}: ${value}` )
```

Comment:
`Object` methods like `keys`, `values`, `entries` as well as `for...in` shouldn't be used as these do not guarantee any particular order of keys/values.
List properties explicitly.
Also remember that that parent class shouldn't know anything about its children props. Child classes must have good reasons to assign or refer parents' class props. There are no such reasons in this task.
Method overloading is to the rescue when building an inhabitant's presentation string.

Reply:
@OleksiyRudenko, but you said "this.whatever looks a bit wordy". 
Is it correct that I must implement toString() at all ending classes (dog, cat, man, woman)?

Reply:
> @OleksiyRudenko, but you said "this.whatever looks a bit wordy".

Hint: `['propAname', 'propBname'].map(propName => this[propName])` -- `this` is referred only once.

> Is it correct that I must implement toString() at all ending classes (dog, cat, man, woman)?

Make benefit from method overloading:
- Each class has its own version of `toString` method to build a string from properties it **owns**
- Every child class inherits it
- If a child class has a unique property it can implement its own version of `toString` that will call parent's version of `toString` (by calling it as `super.toString()`) and supplement the result with values of child class' properties